DeveloperDocs.md

```diff
@@ -7,25 +7,95 @@
 ```
 pip install .[dev]
 ```
-* Copy the file `.secrets/dtenv_template.sh` to `.secrets/dtenv.sh` if you will be using a cloud-based database, or `.secrets/dtenv_localdb.sh` if you will run a local postgres server using docker, and populate this file with values for the various environment variables (ask an existing developer for these).   Note that if using the docker option, you will need to have Docker installed and running on your machine.
 
-### Running the backend API
+## Running an Instance of DTBase
+
+DTBase has three main components:
+
+- Postgresql Database
+- Backend API
+- Frontend API
+
+All these components can be run either locally or deployed via Azure. The following sections contain instructions for running an instance of dtbase either locally or via Azure.
+
+### Local Deployment
+
+#### Running a Local Database
+
+The postgresql database can be run via a locally deployed container with Docker.
+
+1. Copy the file `.secrets/dtenv_template.sh` to `.secrets/dtenv_localdb.sh`
+2. Populate this file with following variables:
+
+```
+#!/bin/bash
+
+# Test database
+export DT_SQL_TESTUSER="postgres"
+export DT_SQL_TESTPASS="password"
+export DT_SQL_TESTHOST="localhost"
+export DT_SQL_TESTPORT="5432"
+export DT_SQL_TESTDBNAME="dt_dev"
+
+# Dev database
+export DT_SQL_USER="postgres"
+export DT_SQL_PASS="password"
+export DT_SQL_HOST="localhost"
+export DT_SQL_PORT="5432"
+export DT_SQL_DBNAME="dt_dev"
+```
+3. Run `source .secrets/dtenv_localdb.sh`
+4. Install Docker
+5. Install postgresql
+6. Run a postgresql server in a docker container:
+
+`docker run --name dt_dev -e POSTGRES_PASSWORD=password -p 5432:5432 -d postgres`
+
+7. To run the backend, we first need to create an empty database:
+
+`createdb --host localhost --username postgres dt_dev`
+
+#### Running the tests
+
+1. Tests can now be run by locally by navigating into dtbase: `cd dtbase` and running the tests: `pytest tests`
+
+#### Running the backend API
+
+The backend API is a flask app that provides REST API endpoints to facilitate reading and writing to the database.
+1. Navigate to the directory `dtbase/backend` and run the command `./run_localdb.sh`. You should then have the flask app listening on `http://localhost:5000` and be able to send HTTP requests to it.  See the [API docs](dtbase/backend/README.md) for details.
+
+#### Running the frontend API
+
+The DTBase frontend is currently an extremely lightweight Flask webapp:
+1. Install npm
+2. Navigate to the directory `dtbase/backend` and run the command `./run.sh`.
+3. You should then have the flask app listening on `http://localhost:5000` and be able to send HTTP requests to it.  See the [API docs](dtbase/backend/README.md) for details.
+
+
+### Running an Azure Deployed Instance of DTBase
```

Comment:
Could we call this "Running with an Azure Deployed Database" or something like that? I'm anticipating another set of docs for running the whole stack, backend and frontend too, on Azure, to be added later.